integration_tests/src/main/python/delta_lake_write_test.py

```diff
@@ -0,0 +1,633 @@
+        cpu_json_lines = cpu_json_data.splitlines()
+        gpu_json_lines = gpu_json_data.splitlines()
+        assert len(cpu_json_lines) == len(gpu_json_lines), "Different line counts in {}:\nCPU: {}\nGPU: {}".format(file, cpu_json_data, gpu_json_data)
+        for c_line, g_line in zip(cpu_json_lines, gpu_json_lines):
+            cpu_json = json.loads(c_line)
+            gpu_json = json.loads(g_line)
+            assert_delta_log_json_equivalent(file, cpu_json, gpu_json)
+
+@allow_non_gpu("ExecutedCommandExec", *delta_meta_allow)
+@delta_lake
+@ignore_order
+@pytest.mark.parametrize("disable_conf",
+                         [{"spark.rapids.sql.format.delta.write.enabled": "false"},
+                          {"spark.rapids.sql.format.parquet.enabled": "false"},
+                          {"spark.rapids.sql.format.parquet.write.enabled": "false"},
+                          {}  # verify disabled by default
+                          ], ids=idfn)
+@pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")
+def test_delta_write_disabled_fallback(spark_tmp_path, disable_conf):
+    data_path = spark_tmp_path + "/DELTA_DATA"
+    assert_gpu_fallback_write(
+        lambda spark, path: unary_op_df(spark, int_gen).coalesce(1).write.format("delta").save(path),
+        lambda spark, path: spark.read.format("delta").load(path),
+        data_path,
+        "ExecutedCommandExec",
+        conf=copy_and_update(writer_confs, disable_conf))
+
+@allow_non_gpu(*delta_meta_allow)
+@delta_lake
+@ignore_order(local=True)
+@pytest.mark.parametrize("gens", parquet_write_gens_list, ids=idfn)
+@pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")
+def test_delta_write_round_trip_unmanaged(spark_tmp_path, gens):
+    gen_list = [("c" + str(i), gen) for i, gen in enumerate(gens)]
+    data_path = spark_tmp_path + "/DELTA_DATA"
+    assert_gpu_and_cpu_writes_are_equal_collect(
+        lambda spark, path: gen_df(spark, gen_list).coalesce(1).write.format("delta").save(path),
+        lambda spark, path: spark.read.format("delta").load(path),
+        data_path,
+        conf=copy_and_update(writer_confs, delta_writes_enabled_conf))
+    with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+
+@allow_non_gpu(*delta_meta_allow)
+@delta_lake
+@ignore_order
+@pytest.mark.parametrize("gens", parquet_part_write_gens, ids=idfn)
+@pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")
+def test_delta_part_write_round_trip_unmanaged(spark_tmp_path, gens):
+    gen_list = [("a", RepeatSeqGen(gens, 10)), ("b", gens)]
+    data_path = spark_tmp_path + "/DELTA_DATA"
+    assert_gpu_and_cpu_writes_are_equal_collect(
+        lambda spark, path: gen_df(spark, gen_list).coalesce(1).write.format("delta")
+            .partitionBy("a")
```

Comment:
is there anything delta specific such that we should have a test with multiple partition columns?

Reply:
I don't know of anything delta-specific regarding multiple partition columns, but I can add a test for that.